doc/manual/faq.rst

```diff
@@ -1,110 +1,171 @@
+            with sequential:
+                self.ttl1.on()  # lane2
+                self.ttl2.on()  # lane3
+                self.ttl3.on()  # lane4
+                self.ttl4.on()  # lane5
+                delay(800*us)
+                self.ttl1.off() # lane5
+                self.ttl2.off() # lane6
+                self.ttl3.off() # lane7
+                self.ttl4.off() # lane0
+        self.ttl0.off()  # lane1 -> clashes with the falling edge of ttl_sma,
+                         # which is already at +800us
+
+In most cases, as in this one, it's relatively easy to rearrange the generation of events so that they will be better spread out across SED lanes without sacrificing actual functionality. One possible solution for the above sequence looks like: ::
 
     @kernel
-    def do(self, i):
-        return i
+    def run(self):
+        self.core.reset()
+        self.ttl0.on()     # lane0
+        self.ttl_sma.on()  # lane1
+        self.ttl1.on()     # lane2
+        self.ttl2.on()     # lane3
+        self.ttl3.on()     # lane4
+        self.ttl4.on()     # lane5
+        delay(800*us)
+        self.ttl1.off()    # lane5
+        self.ttl2.off()    # lane6
+        self.ttl3.off()    # lane7
+        self.ttl4.off()    # lane0  (no clash: new timestamp is higher than last)
+        self.ttl_sma.off() # lane1
+        self.ttl0.off()    # lane2
+
+In this case, the :meth:`~artiq.coredevice.ttl.TTLInOut.pulse` is split up into its component :meth:`~artiq.coredevice.ttl.TTLInOut.on` and  :meth:`~artiq.coredevice.ttl.TTLInOut.off` so that events can be generated more linearly.
 
-create and use variable lengths arrays in kernels?
---------------------------------------------------
+organize datasets in folders?
+-----------------------------
 
-Don't. Preallocate everything. Or chunk it and e.g. read 100 events per
-function call, push them upstream and retry until the gate time closes.
+Use the dot (".") in dataset names to separate folders. The GUI will automatically create and delete folders in the dataset tree display.
 
-execute multiple slow controller RPCs in parallel without losing time?
-----------------------------------------------------------------------
+organize experiment windows in the dashboard?
+---------------------------------------------
+
+Experiment windows can be organized by using the following hotkeys:
 
-Use ``threading.Thread``: portable, fast, simple for one-shot calls.
+* CTRL+SHIFT+T to tile experiment windows
+* CTRL+SHIFT+C to cascade experiment windows
+
+The windows will be organized in the order they were last interacted with.
+
+create and use variable-length arrays in kernels?
+-------------------------------------------------
+
+Don't. Preallocate everything. Or chunk it and e.g. read 100 events per function call, push them upstream and retry until the gate time closes. See also the corresponding notes under :ref:`compiler-types`.
```

Comment:
``Preallocate everything, or chunk it and read a small portion (e.g. 100 events) ...``